src/state.rs

```diff
@@ -250,6 +255,60 @@ impl ProxyState {
         }
         None
     }
+
+    fn load_balance<'a>(&self, src: &Workload, svc: &'a Service) -> Option<&'a Endpoint> {
```

Comment:
this seems to ignore failover/strict and always does a best match, yeah?

Reply:
```
                        // Doesn't match all, and required to. Do not select this endpoint
                        if lb.mode == LoadBalancerMode::Strict && rank != lb.targets.len() {
                            return None;
                        }
```

should do it